test/com/amazon/ion/impl/IonReaderContinuableTopLevelBinaryTest.java

```diff
@@ -2568,25 +2605,24 @@ public void deeplyNestedValueNotOversizedDueToContainerHeaderLengthsNonIncrement
         );
         readerBuilder = readerBuilder.withIncrementalReadingEnabled(false);
         reader = boundedReaderFor(new ByteArrayInputStream(bytes), 5, 5, byteAndOversizedValueCountingHandler);
-        nextExpect(IonType.STRING);
-        expectString("12");
-        nextExpect(IonType.SEXP);
-        stepIn();
-        nextExpect(IonType.SEXP);
-        stepIn();
-        nextExpect(IonType.SEXP);
-        stepIn();
+        assertSequence(
+            next(IonType.STRING), stringValue("12"),
+            next(IonType.SEXP), STEP_IN,
+                next(IonType.SEXP), STEP_IN,
+                    next(IonType.SEXP), STEP_IN
+        );
         expectOversized(0);
-        nextExpect(IonType.STRING);
+        assertSequence(next(IonType.STRING));
         expectOversized(0);
```

Comment:
Can the oversized counter can be decremented? Otherwise, is it just as strong to assert 0 at the end so that you don't have to break this into two sequences?

Reply:
Yes, I'm guessing I added assertions at several points during development while trying to get this to pass, and it survived the restructuring. I'll simplify into one assertion sequence with a single assertion at the end.